process/transaction/shardProcess.go

```diff
@@ -256,6 +256,8 @@ func (txProc *txProcessor) executingFailedTransaction(
 		return nil
 	}
 
+	//TODO: check here, there is no  if !flagPenalizedTooMuchGas.IsSet():
+	// could it happen that not all taken fee or too much is taken from the sender account?
```

Comment:
this is correct. this was never changed. it should stay as it is.

Reply:
ok! removed